### PR TITLE
feat(formik): improve formik types

### PR DESCRIPTION
**Pull request by sibelius**
I think FormikHOC is wrong

we need to type `withFormik` to make this work fine

right now I'm using FormikProps instead

something like this

```
type Values = { a: string }
type Props = {} & FormikProps<Values>
```

### CODE REVIEW

**Comment by sibelius** [comment]
this makes sense for `withFormik`:

```
declare export var withFormik: (
    Component: React$ComponentType<Props>
  ): React$ComponentType<$Diff<Props, ...FormikProps<*>>
```

I've followed this https://flow.org/en/docs/react/hoc/

**Comment by sibelius** [comment]
improved here

```jsx
declare export function withFormik<Props, Values>({
    mapPropsToValues: (props: Props) => Values,
    validationSchema: any, // TODO - improve this
  }): (ComponentType<Props>) => ComponentType<$Diff<Props, FormikProps<Values>>>;
```

How can I make sure this is working fine?

How can I check the type of a single variable for instance?

File: definitions/npm/formik_v0.11.x/flow_v0.53.x-/formik_v0.11.x.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+     *       const {setFieldValue, status, errors, dirty} = this.props;
+     *     }
+     *  }
+     *
+     *
+     * <MyFormikHoc/>
+    */
+   declare export type FormikHOC = {
+    dirty?: $PropertyType<FormikComputedProps<*>, 'dirty'>,
+    errors?: FormikErrors<Values>,
```
Comment (GAntoine) [inline definitions/npm/formik_v0.11.x/flow_v0.53.x-/formik_v0.11.x.js]: `Values` is undefined here, and is failing the specs.

File: definitions/npm/formik_v0.11.x/flow_v0.53.x-/formik_v0.11.x.js [hunk 2]
```diff
@@ -1,3 +1,6 @@
+import { ComponentType } from "react";
+import { FormikProps } from "formik";
```
Comment (saadq) [inline definitions/npm/formik_v0.11.x/flow_v0.53.x-/formik_v0.11.x.js]: I don't think this currently works, see discussion [here](https://github.com/flowtype/flow-typed/issues/1857).

Comment (sibelius) [reply definitions/npm/formik_v0.11.x/flow_v0.53.x-/formik_v0.11.x.js]: this is wrong I guess

**Comment by sibelius** [comment]
@GAntoine could u review again

**Pull request merged**